udata/features/transfer/notifications.py

```diff
@@ -1,11 +1,107 @@
 import logging
 
+from udata.api_fields import field, generate_fields
+from udata.core.dataservices.models import Dataservice
+from udata.core.dataset.models import Dataset
+from udata.core.organization.models import Organization
+from udata.core.reuse.models import Reuse
+from udata.core.user.models import User
 from udata.features.notifications.actions import notifier
 from udata.models import Transfer
+from udata.mongo import db
 
 log = logging.getLogger(__name__)
 
 
+@generate_fields()
+class TransferRequestNotificationDetails(db.EmbeddedDocument):
+    transfer_owner = field(
+        db.GenericReferenceField(choices=(User, Organization), required=True),
+        readonly=True,
+        auditable=False,
+        allow_null=True,
+        filterable={},
+    )
+    transfer_recipient = field(
+        db.GenericReferenceField(choices=(User, Organization), required=True),
+        readonly=True,
+        auditable=False,
+        allow_null=True,
+        filterable={},
+    )
+    transfer_subject = field(
+        db.GenericReferenceField(choices=(Dataset, Dataservice, Reuse), required=True),
+        readonly=True,
+        auditable=False,
+        allow_null=True,
+        filterable={},
+    )
+
+
+@Transfer.on_create.connect
+def on_transfer_created(transfer, **kwargs):
+    """Create notification when a new transfer request is created"""
+
+    from udata.features.notifications.models import Notification
+
+    recipient = transfer.recipient
+    owner = transfer.owner
+
+    if isinstance(recipient, User):
+        try:
+            existing = Notification.objects(
+                user=recipient,
+                details__transfer_recipient=recipient,
+                details__transfer_owner=owner,
+                details__transfer_subject=transfer.subject,
+            ).first()
+
+            if not existing:
+                notification = Notification(
+                    user=recipient,
+                    details=TransferRequestNotificationDetails(
+                        transfer_owner=owner,
+                        transfer_recipient=recipient,
+                        transfer_subject=transfer,
```

Comment:
`.subject` is missing, no?

```suggestion
                        transfer_subject=transfer.subject,
```